napari/layers/points/points.py

```diff
@@ -644,16 +665,41 @@ def _extent_data(self) -> np.ndarray:
             extrema = np.vstack([mins, maxs])
         return extrema
 
+    @property
+    def out_of_slice_display(self) -> bool:
+        """bool: renders points slightly out of slice."""
+        return self._out_of_slice_display
+
+    @out_of_slice_display.setter
+    def out_of_slice_display(self, out_of_slice_display: bool) -> None:
+        self._out_of_slice_display = bool(out_of_slice_display)
+        self.events.out_of_slice_display()
+        self.refresh()
+
     @property
     def n_dimensional(self) -> bool:
-        """bool: renders points as n-dimensional."""
-        return self._n_dimensional
+        warnings.warn(
+            trans._(
+                "'n_dimensional' is deprecated and will be removed in napari v0.5.0, "
+                "use 'out_of_slice_display' instead."
+            ),
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return self._out_of_slice_display
 
     @n_dimensional.setter
-    def n_dimensional(self, n_dimensional: bool) -> None:
-        self._n_dimensional = n_dimensional
+    def n_dimensional(self, value: bool) -> None:
+        warnings.warn(
+            trans._(
+                "'n_dimensional' is deprecated and will be removed in napari v0.5.0, "
+                "use 'out_of_slice_display' instead."
+            ),
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        self.out_of_slice_display = value
         self.events.n_dimensional()
```

Comment:
Even though it's a bit counterintuitive, I think you want to emit the event in the `out_of_slice_display` setter instead of here. Because if the user is connected to the deprecated `n_dimensional` event, it will still get triggered if they change `out_of_slice_display` (because that will cause the value of `self.n_dimensional` to also be different).

Reply:
Sorry, that's terribly worded. To clarify, I see the event as indicating that the value I would get from evaluating `self.n_dimensional` has changed, which is why the event belongs in the `out_of_slice_display` setter.

Reply:
uhm, I see!

Reply:
Done!